### PR TITLE
#161374570 Connect to travel calendar API

### PR DESCRIPTION
**Pull request by asamisellah**
#### What does this PR do?
Connect the travel calendar API service to the make calls to the travel calendar API implemented in the backend.

#### Description of Task to be completed?
- Make travel calendar API send requests to the travel calendar backend implementation
- Make travel calendar details show by default

##### Prerequisites 
You should have your back-end server running
After Logging in, change your user role to  travel administrator in the database

##### Backend
- Using your terminal, clone this repo -``` git clone https://github.com/andela/travel_tool_back.git```
- Change directory to the repo in your text editor-```cd travela_tool_back```
- Open the project in your text editor
- Run ```git fetch``` on your terminal
- Checkout to this branch-```git checkout ft-see-travel-calender-161374998```
- Run ```yarn install``` on your terminal
- Run ```yarn start:dev``` on your terminal to start the backend server

##### Frontend
- Using your terminal, clone this repo -``` git clone https://github.com/andela/travel_tool_front.git``` 
- Change directory to the repo in your text editor-```cd travela_tool_front```
- Open the project in your text editor
- Run ```git fetch``` on your terminal
- Checkout to this branch-```git checkout ch-connect-travel-calendar-API-161374998```
- Install the dependencies-```yarn install```
- Start the server-```yarn start ```
- Navigate to localhost:3000 in your browser
- Login to the application
- Navigate to the dashboard on the side nav
- if there are no approved requests with travel ticket details, the travel calendar should display the message ``` No records found```

##### Create travel Itinerary
- Make requests of the various types of trips 
- Approve the requests
- Add Travel ticket details in the travel checklist modal
- Navigate to the dashboard on the side nav
- Check the travel calendar section and filter by date to view flight details

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
[#161374998](https://www.pivotaltracker.com/n/projects/2184887/stories/161374998)

#### Screenshots (if appropriate)
screen shot 2018-11-23 at 14 37 52

#### Questions:
None

### CODE REVIEW

**Comment by janetnim** [comment]
I've added flight details but I still cannot be able to see the details on the card in the dashboard.

File: src/components/TravelCalendar/index.jsx [hunk 1]
```diff
@@ -98,7 +98,7 @@ class TravelCalendar extends Base {
     } else if(travelCalendarError){
       return (
         <div className="demo-card-wide mdl-card mdl-shadow--2dp errorMsg">
-          <p className="errorMsg__text">Records Not Found</p>
+          <p className="errorMsg__text">No records found</p>
```
Comment (DennisWanjiru) [inline src/components/TravelCalendar/index.jsx]: Its advised when you are following BEM naming conventions to NOT use camelCase naming. Kindly change  the class name to `error-msg__text`

**Comment by janetnim** [comment]
I've tested it and it works. Implement the changes requested and add solve conflicts.

File: src/components/TravelCalendarDetails/__tests__/TravelCalendarDetails.test.js [hunk 2]
```diff
@@ -34,8 +35,8 @@ describe('Travel Calendar Details', () => {
   it('should show/hide details when button is clicked', () => {
     const wrapper = shallow(<TravelCalendarDetails calendar={calendar}/>);
     wrapper.find('button').simulate('click');
-    expect(wrapper.find('button').text()).toEqual('Hide Details');
-    wrapper.find('button').simulate('click');
     expect(wrapper.find('button').text()).toEqual('Show Details');
+    wrapper.find('button').simulate('click');
+    expect(wrapper.find('button').text()).toEqual('Hide Details');
   });
 });
```
[the remark this answers is not in the document]
Comment (asamisellah) [reply src/components/TravelCalendarDetails/__tests__/TravelCalendarDetails.test.js]: @janetnim I have made the requested change. You need to pull the latest changes in the backend branch in order for you to see the changes.

File: src/components/TravelCalendar/index.jsx (hunk 1 above)
Comment (asamisellah) [reply src/components/TravelCalendar/index.jsx]: Thanks for the feedback, I have resolved this

**Comment by amarachukwu-agbo** [comment]
@asamisellah , the storyId in the PR title does not match that on PT board, please correct it.

Also the date range picker does not close when there is a click outside the component like we discussed. Is there a reason why the implementation was changed?

While testing the PR, I noticed that when I select a date range that does not have a record, I still see one record(which does not fall under the selected range) and the div that shows 'No records found'. 

screenshot 2018-11-23 at 6 16 35 pm

**Comment by asamisellah** [comment]
@amarachukwu-agbo I'm on it

**Comment by asamisellah** [comment]
@amarachukwu-agbo About the calendar, we agreed that for the moment I make the calendar close once the user has selected a range of dates, just as Dennis had implemented in the analytics calendar. That is working as expected.

**Pull request merged**